### ISSUE TITLE
Pagination: since parameter is wrong or unclear

### ISSUE THREAD

**Issue by bmenant** [ulule/developers.ulule.com#22]
First, given my experimentation with the API, the preferred pagination offered by the API through the `meta` properties works with `limit` and `offset`. It works very well, but it is not documented.

Nevertheless, if I follow what the documentation says, `GET https://api.ulule.com/v1/projects/{id}/orders?limit=20&since=1525132800` returns

```js
{
  "meta": {
    "limit":20,
    "next":"?limit=20&since=1525097486"
  },
  "orders": [
    {
      // ...
      created_at: "2018-04-30T20:49:19.587376Z",
      // ...
    },
    {
      // ...
      created_at: "2018-04-30T20:26:14.19268Z",
      // ...
    },
    // ...
  ]
}
```

Notes:
------- 

- The requested unix timestamp`1525132800` is equivalent to `2018-05-01T00:00:00+00:00`.
- Returned items are anti-chronologically ordered (i.e. the first record in the list is the youngest, the last one is the oldest).
- _since_ commonly stands for [“from then till now”](https://www.wordreference.com/definition/since).

Issues or miss-expectations:
------

- `next` meta URL gives an anterior `since` parameter value: `1525132800` > `1525097486`. It is logically surprising, as a prior datetime for a _since_ parameter, together with reverse chronological order should start with the same records and only include more records .
- Returned records have been created before the given datetime: `2018-05-01T00:00:00+00:00` > `2018-04-30T20:49:19.587376Z` > `2018-04-30T20:26:14.19268Z`... It means they have **not** been created from then (2018-05-01T00:00:00+00:00) till now, but **before** then.

As a matter of fact, IMHO the `since` parameter erroneously behaves like `until` and should be renamed accordingly or deprecated or whatever you feel appropriated.

**Comment by yansal** [comment]
Thank you for opening this issue, this is greatly appreciated.

Indeed, Ulule API supports offset-based pagination but it's not documented because we're not sure we want to continue to support it in the future, at least for list endpoints such as https://developers.ulule.com/#list-project-orders. Please avoid to develop programs that work with it, as it may become unsupported.

Regarding the wording of the `since` parameter, we agree that it's a misnomer. Because we care a lot about backward compatibility, we will continue to support it.

It would be OK to introduce a better word. `until` may be a better word in this case, but let's consider we introduce a parameter to change the order of the list, allowing users to list orders in the chronological order. `until` would become a misnomer just like `since` is today. This name of this parameter should really mean "this is a time-based offset". `offset` is already taken, although undocumented. Maybe `after` could be a good name. Do not hesitate to share ideas of improvement here. However, please note that this is a low-priority issue for us.

**Comment by bmenant** [comment]
Disclaimer: I didn't request the last version of the API. It might be an old issue that has been solved already. I'll check it out later on today.

**Comment by bmenant** [comment]
> but let's consider we introduce a parameter to change the order of the list, allowing users to list orders in the chronological order. until would become a misnomer just like since is today.

True. This is why, as an API consumer, I’m not a huge fan of time based offset parameters when it comes to paginates results. Time based parameters are great as filters though. IMHO, pagination should only provides means to browse a given set of records (i.e. an actual filtering and sorting configuration request). 

Then, for time sequenced list of records, it might be unclear if `after` references a point in time or a cursor within a set. I would prefer `created_after` over `after`: “fetch all records created _after_ this particular one (or datetime).”

Conventional `limit` and `offset` are actually good _strict_ pagination parameters. However, I understand it might be tough to support such conventional parameters when you deal with highly dynamic and large set of events. In such cases, I’m not sure it makes sense to speak of _pagination_ anyway to avoid all confusion (the worst I have been given to see is a tricky combination of `until`, `since`, `after` and `before` parameters, each of them having their very own behavior).
